### DISCUSSION TITLE
expected undefined to equal 200 cypress

### DISCUSSION THREAD

**Discussion by CarlosLins1** [cypress-io/cypress#17907] [Questions and Help]
alguém pode me dizer onde está o problema nessa validação do status code, quando troco o cy.server() e cy.route() pelo cy.intercept o teste retorna o erro do título.

testando com a versão 4.1 do cypress usando o cy.route, funciona corretamente
```
cy.server();
        cy.route('POST', '**/ongs').as('postOng');
        cy.get('[data-cy=submit]').click();
        cy.wait('@postOng').then((xhr) => {
            expect(xhr.status).be.eq(200);
            expect(xhr.response.body).has.property('id');
            expect(xhr.response.body.id).is.not.null;
            });
    });
```
![image](https://user-images.githubusercontent.com/89609258/131029758-64189bf7-1d49-4707-b4fe-18fc5def8ff1.png)

testando com a versão 8.3 do cypress usando o cy.intercept a validação não funciona
```
 cy.intercept('POST', '**/ongs').as('postOng');
        cy.get('[data-cy=submit]').click();
        cy.wait('@postOng').then((xhr) => {
            expect(xhr.status).be.eq(200);
            expect(xhr.response.body).has.property('id');
            expect(xhr.response.body.id).is.not.null;
        });

    });
```
![image](https://user-images.githubusercontent.com/89609258/131029803-ff754e92-8175-458e-b331-5613d4552d5f.png)

**Comment by CarlosLins1** [comment]
resolvido, alterando para `expect(xhr.response.statusCode).be.eq(200);`

**Comment by tit** [comment]
https://docs.cypress.io/api/commands/intercept#Using-the-yielded-object

```
expect(xhr.statusCode).be.eq(200);
```
